src/main/java/my/test/demo/DownloadDataDemo.java

```diff
@@ -0,0 +1,45 @@
+package my.test.demo;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.FluentWait;
+
+import java.time.Duration;
+
+public class DownloadDataDemo {
+    private final WebDriver driver;
+
+    public DownloadDataDemo(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public boolean downloadData() {
+        driver.navigate().to("https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html");
+        WebElement downloadButton = driver.findElement(By.xpath("//button[@id='cricle-btn']"));
+        downloadButton.click();
+
+        FluentWait<WebDriver> wait = new FluentWait<>(driver)
+                .withTimeout(Duration.ofMinutes(1))
+                .pollingEvery(Duration.ofSeconds(1))
+                .ignoring(NoSuchElementException.class);
+
+        wait.until(driver -> {
+            WebElement percentageOfDownloading1 = driver.findElement(By.xpath("//div[@class='percenttext']"));
+            String getTextOnPage = percentageOfDownloading1.getText();
+            if (getTextOnPage.equals("50%")) {
```

Comment:
Percentage must be >= 50 according to task. This code will not work in case if download process will be fast. Please update